src/libcore/tests/iter.rs

```diff
@@ -1403,6 +1403,7 @@ fn test_repeat() {
     assert_eq!(it.next(), Some(42));
     assert_eq!(it.next(), Some(42));
     assert_eq!(it.next(), Some(42));
+    assert_eq!(it.nth(usize::MAX), Some(42));
```

Comment:
Consider doing this with something that's need_drop, since LLVM already can remove the loop for trivial things like i32: https://play.rust-lang.org/?gist=546bdfadcb744e3aadabf698267b204f&version=nightly

Reply:
Are you sure the loop is being removed here? https://play.rust-lang.org/?gist=c93f5f86a61ea6af7db1b3c2ab22b66f&version=stable is terminated for taking too long to execute (which was the intention for this test case as well), which isn't the case after the specialisation.

Reply:
Try https://play.rust-lang.org/?gist=c93f5f86a61ea6af7db1b3c2ab22b66f&version=stable&mode=release, and it doesn't time out.  (Now, I'm sure at least one of the builds runs the test in debug, so it'd still catch it, but it's also a Copy type, so copy elision is definitely allowed -- a dead memcpy is certainly removable -- compared to a more complex Clone impl that's the subject of discussion in the issue.)

Reply:
Ah, got it! I'll adjust the test, thanks!